server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/query/QueryTranslator.java

```diff
@@ -164,6 +165,7 @@ static MailAddress parseMailAddress(String mailAddressString) throws QueryTransl
 
     private final ImmutableTable<FieldName, Operator, Function<String, Criterion>> criterionRegistry;
 
+    @Inject
     public QueryTranslator(MailboxId.Factory mailboxIdFactory) {
```

Comment:
should not be public

Reply:
Well, this constructor is accessed bu the route tests located in another package

Moving it to the same package creates much conflicts (DTO package-private constructors)

The only solution might be fusing the two packages. What I don't like.

Reply:
use @VisibleForTesting + public then